File: 0785-is-graph-bipartite/0785-is-graph-bipartite.cpp
class Solution {
public:
    bool bfs(int start,vector<int>&color,vector<vector<int>>&adj){
        
        queue<int>q;
        color[start]=0;
        q.push(start);

        while(!q.empty()){
            int node=q.front();
            q.pop();

            for(int nb:adj[node]){
                
                if(color[nb]==-1){
                    color[nb]=(color[node]+1)%2;
                    q.push(nb);
                }
                else{
                    if(color[node]==color[nb]){
                        return false;
                    }
                }

            }
        }
        return true;

    }
    bool isBipartite(vector<vector<int>>& adj) {
        int n=adj.size();
        vector<int>color(n,-1);

        queue<int>q;
        for(int i=0;i<n;i++){
            if(color[i]==-1){
                if(!bfs(i,color,adj)) return false;
            }
        }
        return true;
        

       
    }
};

File: 0785-is-graph-bipartite/0785-is-graph-bipartite.py
from collections import deque
class Solution:
    def isBipartite(self, graph: List[List[int]]) -> bool:

        n=len(graph)
        color=[-1]*n

        for i in range(n):

            if color[i]==-1:
                q=deque()
                q.append(i)
                color[i]=0

                while q :
                    node =q[0]
                    q.popleft()
                    for nb in graph[node]:

                        if color[nb] == -1:
                            color[nb] = (color[node] + 1) % 2
                            q.append(nb)

                        elif color[nb] == color[node]:
                            return False
        
        return True


        
        